site/src/components/RichParameterInput/RichParameterInput.tsx

```diff
@@ -0,0 +1,207 @@
+    </label>
+  )
+}
+
+export interface RichParameterInputProps {
+  disabled?: boolean
+  parameter: TemplateVersionParameter
+  onChange: (value: string) => void
+}
+
+export const RichParameterInput: FC<RichParameterInputProps> = ({
+  disabled,
+  onChange,
+  parameter,
+  ...props
+}) => {
+  const styles = useStyles()
+
+  return (
+    <Stack direction="column" spacing={0.75}>
+      <ParameterLabel parameter={parameter} />
+      <div className={styles.input}>
+        <RichParameterField
+          {...props}
+          disabled={disabled}
+          onChange={onChange}
+          parameter={parameter}
+        />
+      </div>
+    </Stack>
+  )
+}
+
+const RichParameterField: React.FC<RichParameterInputProps> = ({
+  disabled,
+  onChange,
+  parameter,
+  ...props
+}) => {
+  const [parameterValue, setParameterValue] = useState(parameter.default_value)
+  const styles = useStyles();
+
+  if (isBoolean(parameter)) {
+    return (
+      <RadioGroup
+        defaultValue={parameter.default_value}
+        onChange={(event) => {
+          onChange(event.target.value)
+        }}
+      >
+        <FormControlLabel
+          disabled={disabled}
+          value="true"
+          control={<Radio color="primary" size="small" disableRipple />}
+          label="True"
+        />
+        <FormControlLabel
+          disabled={disabled}
+          value="false"
+          control={<Radio color="primary" size="small" disableRipple />}
+          label="False"
+        />
+      </RadioGroup>
+    )
+  }
+
+  if (parameter.options.length > 0) {
+    return (
+      <RadioGroup
+        defaultValue={parameter.options[0].value}
+        onChange={(event) => {
+          onChange(event.target.value)
+        }}
+      >
+        {
+          parameter.options.map((option) => (
+            <FormControlLabel
+              disabled={disabled}
+              key={option.name}
+              value={option.value}
+              control={<Radio color="primary" size="small" disableRipple />}
+              label={(
+                <span>
+                  {option.icon && (
+                    <img
+                      className={styles.optionIcon}
+                      alt="Parameter icon"
+                      src={option.icon}
+                      style={{
+                        pointerEvents: "none",
+                      }}
+                    />
+                  )}
+                  {option.name}
+                </span>
+              )}
+            />
+          ))
+        }
+      </RadioGroup>
+    )
+  }
+
+  // A text field can technically handle all cases!
+  // As other cases become more prominent (like filtering for numbers),
+  // we should break this out into more finely scoped input fields.
+  return (
+    <TextField
+      {...props}
+      type={parameter.type}
+      size="small"
+      disabled={disabled}
+      placeholder={parameter.default_value}
+      value={parameterValue}
```

Comment:
This error usually happens when the value changes from `string` to `undefined`. Many times, having undefined values is a "bug" warning but in a few cases, it is ok so what you have to do is pass an empty string as fallback: 

```
value={parameterValue ?? ""}
```

Reply:
I introduced the `parameterValue` as I wasn't able to fix it with formik. It looks like formik has troubles with understanding the concept of `array of object`, and for some reason, the input goes undefined. I will look into this.

Reply:
I see, so maybe the `getFormHelpers` has a bug 🤔